### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by flip1995**
I'll write some TODOs later. This time I'll need a review for this Rustup

r? @ghost  (for now)

changelog: none

### CODE REVIEW

**Comment by xFrednet** [comment]
Even with the state of rust-lang/rust#88401 being unclear (A PR reverting rust-lang/rust#80357), I think we should move forward with this rustup. Fingers are still crossed that the bug will be fixed and nothing has to be reverted. In the worst case, we can try to not revert everything in Clippy but only adjust the parsing of effected expressions in [`clippy_utils::higher`](https://github.com/rust-lang/rust-clippy/blob/0828c0b06170d943579dbd41c19b2073471bdecc/clippy_utils/src/higher.rs).

I'll finish my review and then see if any feedback on this has been added. If not I'll leave another comment postponing the merge a bit more, as I would like to not make this decision alone :sweat_smile: 

cc: @rust-lang/clippy

**Comment by flip1995** [comment]
I would wait for the revert to get figured out first for at least next Thursday. There's no important fixes we have to sync back to rustc AFAIK. So moving forward with the sync isn't that important right now and can be delayed for a few days IMO.

I really don't want to take the gamble of having to revert everything again. Especially since I don't know how it will interact with the `git-subtree` tool.

File: clippy_lints/src/utils/internal_lints/metadata_collector.rs [hunk 1]
```diff
@@ -344,16 +344,11 @@ fn parse_config_field_doc(doc_comment: &str) -> Option<(Vec<String>, String)> {
...
+            let documentation = doc_comment.split_off(split_pos);
 
-            // Extract lints
             doc_comment.make_ascii_lowercase();
             let lints: Vec<String> = doc_comment.split_off(DOC_START.len()).split(", ").map(str::to_string).collect();
 
-            // Format documentation correctly
-            // split off leading `.` from lint name list and indent for correct formatting
-            documentation = documentation.trim_start_matches('.').trim().replace("\n ", "\n    ");
-
```
[the remark this answers is not in the document]
Comment (xFrednet) [reply clippy_lints/src/utils/internal_lints/metadata_collector.rs]: I've reverted this change as it messed with the formatting and I couldn't find a reason why this was done

**Review by xFrednet — APPROVED** [review]
I've now reviewed all commits that have been done as part of this rustup/merge and pushed a small fix. This should be good to go from my point of view. However, these changes are also dependent on changes in the Rust repo. I'll therefore only approve it now and not r+ it.

**Comment by flip1995** [comment]
It seems like rust-lang/rust#88572 fixes the issue and a revert won't be necessary, so I'm going through with this rustup now.

**Comment by flip1995** [comment]
Thanks for merging. I just managed to push it up and then had to jump on a plane. I'll do the Rust side of the sync first thing tomorrow.

**Pull request merged**